src/WebJobs.Script/Utility.cs

```diff
@@ -169,6 +169,43 @@ internal static async Task InvokeWithRetriesWhenAsync(Func<Task> action, int max
         internal static Task InvokeWithRetriesAsync(Func<Task> action, int maxRetries, TimeSpan retryInterval)
             => InvokeWithRetriesWhenAsync(action, maxRetries, retryInterval, (e) => !e.IsFatal());
 
+        /// <summary>
+        /// Invokes an asynchronous action with retry semantics using exponential backoff between attempts.
+        /// Only non-fatal exceptions (see IsFatal extension) are retried. The final (exhausted) failure is rethrown.
+        /// </summary>
+        /// <param name="action">The asynchronous action to execute.</param>
+        /// <param name="maxRetries">Maximum number of retries (not counting the initial attempt).</param>
+        /// <param name="logger">Optional logger for warning messages and delay diagnostics.</param>
+        /// <param name="unit">Base time unit for backoff (defaults to 1 second).</param>
+        /// <param name="min">Minimum delay enforced for backoff.</param>
+        /// <param name="max">Maximum delay enforced for backoff.</param>
+        /// <param name="cancellationToken">Cancellation token controlling delay waits.</param>
+        internal static async Task InvokeWithRetriesBackoffAsync(
+            Func<Task> action,
+            int maxRetries,
+            ILogger logger = null,
```

Comment:
Consider having a BackoffOptions entity to describe these values.

Reply:
Also consider optional jitters (A random value to avoid overwhelming the system, a recommendition for distributed systems)